src/lang/std/engineConnection.ts

```diff
@@ -314,223 +571,182 @@ class EngineConnection {
           }
-        }
-      } else if (resp.type === 'trickle_ice') {
-        let candidate = resp.data?.candidate
-        this.pc?.addIceCandidate(candidate as RTCIceCandidateInit)
-      } else if (resp.type === 'ice_server_info' && this.pc) {
-        console.log('received ice_server_info')
-        let ice_servers = resp.data?.ice_servers
-
-        if (ice_servers?.length > 0) {
+
           // When we set the Configuration, we want to always force
           // iceTransportPolicy to 'relay', since we know the topology
           // of the ICE/STUN/TUN server and the engine. We don't wish to
           // talk to the engine in any configuration /other/ than relay
           // from a infra POV.
-          this.pc.setConfiguration({
+          this.pc?.setConfiguration({
             iceServers: ice_servers,
             iceTransportPolicy: 'relay',
           })
-        } else {
-          this.pc?.setConfiguration({})
-        }
 
-        // We have an ICE Servers set now. We just setConfiguration, so let's
-        // start adding things we care about to the PeerConnection and let
-        // ICE negotiation happen in the background. Everything from here
-        // until the end of this function is setup of our end of the
-        // PeerConnection and waiting for events to fire our callbacks.
-
-        this.pc.addEventListener('icecandidate', (event) => {
-          if (!this.pc || !this.websocket) return
-          if (event.candidate !== null) {
-            console.log('sending trickle ice candidate')
-            const { candidate } = event
-            this.send({
-              type: 'trickle_ice',
-              candidate: candidate.toJSON(),
-            })
+          this.state = {
+            type: EngineConnectionStateType.Connecting,
+            value: {
+              type: ConnectingType.ICEServersSet,
+            },
           }
-        })
 
-        // Offer to receive 1 video track
-        this.pc.addTransceiver('video', {})
-
-        // Finally (but actually firstly!), to kick things off, we're going to
-        // generate our SDP, set it on our PeerConnection, and let the server
-        // know about our capabilities.
-        this.pc
-          .createOffer()
-          .then(async (descriptionInit) => {
-            await this?.pc?.setLocalDescription(descriptionInit)
-            console.log('sent sdp_offer begin')
-            this.send({
-              type: 'sdp_offer',
-              offer: this.pc?.localDescription,
+          // We have an ICE Servers set now. We just setConfiguration, so let's
+          // start adding things we care about to the PeerConnection and let
+          // ICE negotiation happen in the background. Everything from here
+          // until the end of this function is setup of our end of the
+          // PeerConnection and waiting for events to fire our callbacks.
+
+          // Add a transceiver to our SDP offer
+          this.pc?.addTransceiver('video', {})
```

Comment:
No problem to change it, the well-of-motivation runs deep in this vessel. :fire:

Reply:
So in browsers, you need to use `addTransceiver(kind, config)` to signal during `createOffer` you want video. Since we're going one way, we can tell WebRTC it's a one-way via the `config` parameter :)